### ISSUE TITLE
[Build] Question -- is it possible to override/replace a core operator?

### ISSUE THREAD

**Issue by bluejack** [microsoft/onnxruntime#23007]
### Describe the issue

I have read the extensions guidelines and it's not clear. Also, building and linking a shared library is not a trivial project. Before I go too far down the road, I want to see if anyone has done this, and if anyone knows of other examples of this.

### Target platform

arm, but eventually all

### Build script

I'm not quite sure even how to build the shared library yet. 

### Error / output

I tried a straight up gcc build, but am still working out the build flags. presumably there is a cmake example out there somewhere that would be a better idea. 

### GCC / Compiler Version

apple clang 15.0.0

**Comment by skottmckay** [comment]
Yes. If you create a custom operator and register it using the matching domain and name it will override.

https://github.com/microsoft/onnxruntime/blob/549cbc7e69433433cd464efcb6bc78e796fbae3b/onnxruntime/test/framework/local_kernel_registry_test.cc#L223-L235
